qt/widgets/common/src/FunctionModelDataset.cpp

```diff
@@ -0,0 +1,37 @@
+// Mantid Repository : https://github.com/mantidproject/mantid
+//
+// Copyright &copy; 2020 ISIS Rutherford Appleton Laboratory UKRI,
+//   NScD Oak Ridge National Laboratory, European Spallation Source,
+//   Institut Laue - Langevin & CSNS, Institute of High Energy Physics, CAS
+// SPDX - License - Identifier: GPL - 3.0 +
+#include "MantidQtWidgets/Common/FunctionModelDataset.h"
+
+#include <stdexcept>
+
+namespace MantidQt {
+namespace MantidWidgets {
+
+Dataset::Dataset(const QString &workspaceName, const Spectra &spectra)
```

Comment:
As we are copying the two parameters in here it's generally accepted that it can be better to either add an r-value reference overload or take by value. We could take the advantage here that QString is copy on write and just add an overload:

```cpp
Dataset::Dataset(QString workspaceName, Spectra &&spectra) : 
  : m_workspaceName(workspaceName), m_spectra(std::move(spectra)) {}
```
to accept an r-value reference of the spectra and it could avoid copying on construction.